### PR TITLE
docs(release-checklist.md): update from rc1 amended processes

### PR DESCRIPTION
**Pull request by vdice**
ref https://github.com/deis/workflow/issues/258 for proposal introducing most of these changes

also note some steps depend on https://github.com/sgoings/deis-workflow-group/pull/4

### CODE REVIEW

**Comment by jchauncey** [comment]
Right now I'm not sure the `CHANGELOG.md` should be generated so far down on the list. For the RC1 release the change log on the rc1 branch is different than the change log on master.

So my thoughts are that before you cut the branch you publish the changelog pr (from last branch commit to HEAD). That gets merged and then you cut the branch. 

This way the changelog is up to date on master and is also reflected in the release branch.

**Comment by vdice** [comment]
@jchauncey I like that idea in the best-case scenario (no hotfixes needed from release testing) but what's the protocol if there are fixes needed?  Perhaps as simple as adding the updated `CHANGELOG` commit on top (or in conjunction with) of the fix commit?  This might require we follow the paradigm of fixing the release issue _in the release branch_ and merging the release branch into master when release is finished...

File: src/roadmap/release-checklist.md [hunk 1]
```diff
@@ -34,47 +36,92 @@ A release consists of the following artifacts:
...
+  make git-update # point all repos to latest master commits
+  BRANCH="release-$WORKFLOW_RELEASE" NEW="true" make git-checkout-branch
+  BRANCH="release-$WORKFLOW_RELEASE" make git-push-branch #(can use DRY_RUN=true)
+  ```
+
+2. tag and push docker images to 'staging' `deisci` org
+  ```
+   TAG="${WORKFLOW_RELEASE}" ORG="deisci" make docker-tag #(can use DRY_RUN=true)
+   TAG="${WORKFLOW_RELEASE}" ORG="deisci" make docker-push #(can use DRY_RUN=true)
+  ```
```
Comment (krancour) [inline src/roadmap/release-checklist.md]: I the preceding lines (52 - 61), it's not clear which project contains these make targets. In fact, looking around, I cannot even find them. More clarity would be better here.

File: src/roadmap/release-checklist.md [hunk 2]
```diff
@@ -34,47 +36,92 @@ A release consists of the following artifacts:
...
+
+# Step 2: Create New Helm Classic Charts
 
 Next, we'll create new [Helm Classic](https://github.com/helm/helm-classic) charts so that we can "stage" a
 version of our release for testing. Here is the current process to do so:
 
 1. Create a new branch in [deis/charts](https://github.com/deis/charts): `git checkout -b release-$WORKFLOW_RELEASE origin/master`
```
Comment (krancour) [inline src/roadmap/release-checklist.md]: Again... `origin` v `upstream` comment applies.

File: src/roadmap/release-checklist.md [hunk 3]
```diff
@@ -34,47 +36,92 @@ A release consists of the following artifacts:
...
-export WORKFLOW_RELEASE_SHORT=<short form of above>
-```
+(For actions below, [sgoings/deis-workflow-group](https://github.com/sgoings/deis-workflow-group) was used)
+
+1. Once the release milestone is cleared of tickets in the workflow component repos, the release branches can be cut.  
+
+  If only a particular repo is ready, navigate to said repo and:
+  ```
+  git checkout master && git pull origin master
+  git checkout -b release-$WORKFLOW_RELEASE && git push origin release-$WORKFLOW_RELEASE
```
Comment (jchauncey) [inline src/roadmap/release-checklist.md]: its funny you say that. i specifically rename my origin to be deis so i dont accidentally do `git push origin master` (even though we have branch protection)

Comment (krancour) [inline src/roadmap/release-checklist.md]: For reference, here's what our own contributing docs say about this. (These were mostly lifted from v1.)

http://docs-v2.readthedocs.io/en/latest/contributing/development-environment/#fork-the-repository

Comment (vdice) [inline src/roadmap/release-checklist.md]: `origin` replaced with `upstream` here and elsewhere; thanks @krancour

File: src/roadmap/release-checklist.md (hunk 1 above)
Comment (vdice) [inline src/roadmap/release-checklist.md]: Indeed -- please see latest for updated clarification...

**Pull request merged**